design/one-pager-stacks-security-isolation.md

```diff
@@ -56,6 +56,33 @@ From the [Kubernetes documentation](https://kubernetes.io/docs/tasks/access-kube
 Therefore, any time a CRD is created, it will also be at the cluster scope, even if instances of the CRD are scoped to a namespace.
 This has caused confusion in the past, so it's important to clear that up here.
 
+#### Discoverability of Scope
+
+Because CRDs are cluster scoped, but our permission model for stacks is both namespaced and global we will
+ need to label CRDs to indicate which are available in a given context. As the stack-installer processes a
+ cluster or namespace stack install it needs to add labels indicating that it's crossplane environment scoped or
+ namespace scoped to a given namespace.
+
+Labels on a CRD:
+    crossplane.{scope=namespace|environment}=true
```

Comment:
> In our model we assume that any Cluster Stack resources are global will be for the
 environment or infrastructure scope, whereas any namespace stack resources will be for the namespace that that stack is installed into.

Why do we need this label? I believe we've ruled out any scenario where a CRD scope does not match it's intended Crossplane scope.

Namespaced scope CRD are for namespaces
Cluster scoped CRDs are for clusters (environment)

(Contingent on https://github.com/crossplaneio/crossplane/pull/1023)

Reply:
scope is not a label that can be searched on, i.e

```
kubectl get crds --selector=crossplane.scope=environment
```